src/hist.jl

```diff
@@ -213,3 +215,125 @@ fit{N,W}(::Type{Histogram}, vs::NTuple{N,AbstractVector}, wv::WeightVec{W}, edge
     append!(Histogram(edges,W,closed), vs, wv)
 fit{N}(::Type{Histogram},vs::NTuple{N,AbstractVector}, wv::WeightVec; closed::Symbol=:right, nbins=sturges(length(vs[1]))) =
     fit(Histogram, vs, wv, histrange(vs,nbins,closed); closed=closed)
+
+
+# Get a suitable high-precision type for the norm of a histogram.
+@generated function norm_type{T, N, E}(h::Histogram{T, N, E})
+    args = [:( eltype(edges[$d]) ) for d = 1:N]
+    quote
+        edges = h.edges
+        norm_type(promote_type(T, $(args...)))
+    end
+end
+
+norm_type{T<:Integer}(::Type{T}) = promote_type(T, Int64)
+norm_type{T<:AbstractFloat}(::Type{T}) = promote_type(T, Float64)
+
+
+@generated function norm{T, N, E}(h::Histogram{T, N, E})
+    quote
+        edges = h.edges
+        weights = h.weights
+        S = norm_type(h)
+        v_0 = 1
+        s_0 = zero(S)
+        @inbounds @nloops(
+            $N, i, weights,
+            d -> begin
+                v_{$N-d+1} = v_{$N-d} * (edges[d][i_d + 1] - edges[d][i_d])
+                s_{$N-d+1} = zero(S)
+            end,
+            d -> begin
+                s_{$N-d} += s_{$N-d+1}
+            end,
+            begin
+                $(Symbol("s_$(N)")) += (@nref $N weights i) * $(Symbol("v_$N"))
+            end
+        )
+        norm(s_0)
+    end
+end
+
+
+function _float_deepcopy(x)
+    y = float(x)
+    is(y, x)? deepcopy(y) : y
+end
+
+
+function float{T, N, E}(h::Histogram{T, N, E})
+    float_weights = float(h.weights)
+    if is(float_weights, h.weights)
+        h
+    else
+        Histogram{eltype(float_weights), N, E}(deepcopy(h.edges), float_weights, h.closed)
+    end
+end
+
+
+# Normalization modes:
+#
+# * `:norm`: Normalize by norm(h). Resulting histogram has norm 1. Operation
+#    is always idempotent. For homogeneous binning, result is the same as with
+#    mode `:pdf`. For inhomogeneous binning, histograms of same data but with
+#    different binning will have different weight values at same coordinates
+#    after normalization.
+# *  `:pdf`: Normalize by sum of weights and bin sizes. Resulting histogram
+#    has norm 1 and represents a PDF. Operation is *not* idempotent for
+#    histograms with inhomogeneous binning. For homogeneous binning, result is
+#    the same as with mode `:norm`. Histograms of same data but with different
+#    binning will always have similar weight values at same coordinates after
+#    normalization.
+# * `:density`: Normalize by bin sizes only. Resulting histogram represents
+#    count density of input and does not have norm 1. Operation is *never*
+#    idempotent. Histograms of same data but with different binning will
+#    always have similar weight values at same coordinates after
+#    normalization.
+# *  `:none`: Leaves histogram unchanged. Useful to simplify code that has to
+#    conditionally apply different modes of normalization.
+#
+# aux_weights may, e.g., be estimated statistical uncertainties.
```

Comment:
Yes, I had kinda planned this, but wasn't sure because StatsBase isn't exactly rich in docstrings. :-)
Added now.